src/js/actions/layers.js

```diff
@@ -206,50 +219,47 @@ define(function (require, exports) {
             continueOnError: true
         });
 
-        // Fetch extension data by a range of layer indexes.
-        // Always start with index 1 because when a document consists only of a background layer (index 0), 
-        // the photoshop action will fail.
-        // And it is safe to ignore ALL bg layers because we don't use extension data on them
-        var nameSpace = global.EXTENSION_DATA_NAMESPACE,
-            numberOfLayers = doc.hasBackgroundLayer ? doc.numberOfLayers + 1 : doc.numberOfLayers,
-            indexRange = _.range(1, startIndex + numberOfLayers),
-            extensionPlayObjects = indexRange.map(function (i) {
-                var layerRef = layerLib.referenceBy.index(i);
-                return layerLib.getExtensionData(docRef, layerRef, nameSpace);
-            }),
+        var extensionPromise;
+        if (doc.hasBackgroundLayer && doc.numberOfLayers === 0 && targetLayers.length === 1) {
```

Comment:
I special-cased this differently than you did previously, @mcilroyc. The reason is that it's fairly tricky to merge the lazy and metadata properties subset into the full properties set. I think this is actually a good, fast way to do it, but if something looks off to you please let me know.